packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect } from "react";
+import { Field } from "react-final-form";
+import styled from "styled-components";
+import createPersistedState from "use-persisted-state";
+import { Button, Toast, theme } from "@socialgouv/react-ui";
+
+import { QuestionLabel } from "../../common/stepStyles";
+import Search from "../../../conventions/Search/Form";
+import { isNotYetProcessed } from "./situation";
+
+// store selected convention in localStorage
+const useConventionState = createPersistedState("convention");
+
+function StepInfoCCn({ form }) {
+  const [ccInfo, setCcInfo] = useConventionState({});
+  useEffect(() => {
+    form.batch(() => {
+      [
+        "contract",
+        "criteria",
+        "missionFormation",
+        "ruptureContratFauteGrave",
+        "propositionCDIFinContrat",
+        "refusSouplesse",
+        "finContratPeriodeDessai",
+        "propositionCDIFindeContrat",
+        "refusCDIFindeContrat",
+        "interruptionFauteGrave",
+        "refusRenouvellementAuto",
+        "typeRemuneration",
+        "salaire",
+        "salaires"
+      ].forEach(key => form.change(key, undefined));
+      form.change("ccn", ccInfo.convention);
+    });
+  }, [ccInfo, form]);
+  return (
+    <>
+      <Field
+        name="ccn"
+        render={({ input, meta: { error } }) => {
+          if (input.value) {
+            return (
+              <>
+                <QuestionLabel>Votre convention collective</QuestionLabel>
+                <p>
+                  {input.value.title}
+                  <br />
+                  {ccInfo.label && `(${ccInfo.label})`}
+                </p>
+
+                <Button
+                  variant="link"
+                  type="button"
+                  onClick={() => setCcInfo({})}
+                >
+                  Changer de convention collective
+                </Button>
+                {error && <StyledToast>{error}</StyledToast>}
+              </>
+            );
+          }
+          return (
+            <>
+              <QuestionLabel>
+                Quelle est votre convention collective ?
+              </QuestionLabel>
+              <P>
+                <strong>* optionnel</strong>, si vous ne connaissez pas votre
+                convention collective, vous pouvez passer à l’étape suivante en
+                cliquant sur le bouton Suivant.
+              </P>
+              <SearchStyled title="" onSelectConvention={setCcInfo} />
+            </>
+          );
+        }}
+      />
+    </>
+  );
+}
+
+StepInfoCCn.validate = values => {
+  const errors = {};
+  if (Object.keys(values).length === 0) {
+    errors.ccn = "Quelle est votre convention collective ?";
+  } else if (values.ccn && isNotYetProcessed(values.ccn.num)) {
+    errors.ccn =
+      "Le code du travail ne prévoit pas une durée précise du préavis de démission. Il prévoit qu'un texte du niveau légal conventionnel, un accord d'entreprise voire un usage en prévoit les durées et modalités. ";
```

Comment:
the error message not reflect the fact that the ccn is not included in our tools

Reply:
C'est le message que m'a demandé de rajouter Jean René si l'ont ne match pas de résultat